user/overriding-packages.rst

```diff
@@ -0,0 +1,282 @@
+This can be done by  **overriding** the package.
+It is accomplished by sourcing an existing workspace, then building a different version of that package again in a new overlay workspace.
+
+Overriding a package is not always possible.
+This page offers tips for avoiding common issues.
+
+.. contents:: Table of Contents
+    :depth: 3
+
+
+How to avoid common issues
+--------------------------
+
+These are good practices to avoid common issues.
+If the advice is too restrictive then see the known issue descriptions for more complex advice.
+
+Use isolated workspaces
+***********************
+
+If you are building a workspace and suspect you may overriding a package from it in the future, then use an isolated workspace.
+This avoids include directory search order issues.
+See :doc:`this documentation <isolated-vs-merged-workspaces>` if you're unsure what an isolated workspace is.
+
+Override every package that depends on the one you want to override
+*******************************************************************
+
+A **leaf package** is one that has no other packages that depend on it.
+Overriding a non-leaf package can be problematic.
+Packages in the underlay were built against the underlay version of the package, but they will be expected to run with the overlay version.
+If their build process stores some information about the non-leaf package, such as an expected ABI, then undefined behavior can happen at runtime.
+
+Problems caused by packages remembering information at build time can be avoided by overriding every package that directly or indirectly depends on the one you actually want to override.
+The group of overridden packages must span all underlays.
+
+Say there are 3 workspaces, **A**, **B** and **C**, where **C** overlays **B** and **B** overlays **A**.
+**A** contains packages ``foo`` and ``baz`` where ``baz`` depends on ``foo``.
+**B** contains packages ``ping`` that depends on ``foo`` and  ``pong`` that depends on ``baz``.
+**C** is the workspace being built.
+If you want to override ``foo`` then you should also override ``baz``, ``ping``, and ``pong``.
+
+Make sure overridden Python packages do not change entry point specifications
+*****************************************************************************
+
+Python packages may have `entry point specifications <https://packaging.python.org/en/latest/specifications/entry-points/>`_ in a ``setup.py`` or ``setup.cfg`` file.
+Make sure the package in the overlay has identical specifications to the version in the underlay, or only adds new ones.
+If any specification has been changed or removed then it may not be possible to override this package.
+
+How to make it easier for your users to override
+------------------------------------------------
+
+This section has advice for package authors about how to make easier for your users to use your package and override it or other pacakges.
```

Comment:
```suggestion
This section has advice for package authors about how to make it easier for your users to use your package and override it in other pacakges.
```

Reply:
I think `or` is correct here. The `or` applies to overriding other packages while this package is also in the overlay or underlay workspace.